dali/python/nvidia/dali/_debug_mode.py

```diff
@@ -358,53 +368,66 @@ def is_primitive_type(x):
 
         return False, 'cpu', data
 
-    def _create_subpipeline(self, op_wrapper, inputs, kwargs):
-        """Creates pipeline wrapper around operator call (only in the first run).
-        Each pipeline uses ExternalSource operators for processing inputs and kwargs.
-        Returns pipeline and classification of data as batch.
-        """
+    @staticmethod
```

Comment:
I cannot add comment to L356-L369, so I will put it here. `_is_mxnet_array` should have the same dedicated case as `_is_torch_tensor`, and the `device` should be inferred from the array type, not set CPU for mxnet and GPU for torch.

Reply:
done